packages/application-shell-connectors/src/components/application-context/application-context.tsx

```diff
@@ -0,0 +1,217 @@
+  project?: TRawProject;
+  projectDataLocale?: string;
+  children: React.ReactNode;
+};
+type ConsumerProps<ExtendedEnvironment extends {}> = {
+  render: (
+    context: TApplicationContext<ExtendedEnvironment> | {}
+  ) => React.ReactNode;
+  children?: never;
+};
+type DefaultMappedProps<ExtendedEnvironment extends {}> = {
+  applicationContext: TApplicationContext<ExtendedEnvironment> | {};
+};
+
+const Context = React.createContext<TApplicationContext<{}> | {}>({});
+
+const defaultTimeZone = moment.tz.guess() || 'Etc/UTC';
+
+// Expose only certain fields as some of them are only meant to
+// be used internally in the AppShell
+const mapUserToApplicationContextUser = (
+  user?: TRawUser
+): TApplicationContextUser | null => {
+  if (!user) return null;
+
+  return {
+    id: user.id,
+    email: user.email,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    // NOTE: this is an alias for the original field `user.language` but it's actually
+    // a locale (language + country).
+    locale: user.language,
+    timeZone: user.timeZone || defaultTimeZone,
+  };
+};
+
+// Expose only certain fields as some of them are only meant to
+// be used internally in the AppShell
+const mapProjectToApplicationContextProject = (
+  project?: TRawProject
+): TApplicationContextProject | null => {
+  if (!project) return null;
+
+  return {
+    key: project.key,
+    version: project.version,
+    name: project.name,
+    countries: project.countries,
+    currencies: project.currencies,
+    languages: project.languages,
+    ownerId: project.owner.id,
+  };
+};
+
+const createApplicationContext: <ExtendedEnvironment extends {}>(
+  environment: ExtendedEnvironment & TApplicationContextEnvironment,
+  user?: TRawUser,
+  project?: TRawProject,
+  projectDataLocale?: string
+) => TApplicationContext<ExtendedEnvironment> = (
+  environment,
+  user,
+  project,
+  projectDataLocale
+) => ({
+  environment,
+  user: mapUserToApplicationContextUser(user),
+  project: mapProjectToApplicationContextProject(project),
+  permissions: project && project.permissions ? project.permissions : null,
+  dataLocale: projectDataLocale || null,
+});
+
+const ApplicationContextProvider = <ExtendedEnvironment extends {}>(
+  props: ProviderProps<ExtendedEnvironment>
+) => (
+  <Context.Provider
+    value={createApplicationContext<ExtendedEnvironment>(
+      props.environment,
+      props.user,
+      props.project,
+      props.projectDataLocale
+    )}
+  >
+    {props.children}
+  </Context.Provider>
+);
+ApplicationContextProvider.displayName = 'ApplicationContextProvider';
+
+const ApplicationContext = <ExtendedEnvironment extends {}>(
+  props: ConsumerProps<ExtendedEnvironment>
+) => (
+  <Context.Consumer>
+    {applicationContext => props.render(applicationContext)}
+  </Context.Consumer>
+);
+ApplicationContext.displayName = 'ApplicationContext';
+
+function withApplicationContext<
+  Props extends {},
+  MappedProps extends {} = {},
```

Comment:
...this are the props from the `mapApplicationContextToProps`, in case it's used